.circleci/config.yml

```diff
@@ -15,37 +21,45 @@ jobs:
 
             - run: yarn install
 
+            # cache our yarn cache
             - save_cache:
                 key: yarn-v1-{{ checksum "yarn.lock" }}-{{ arch }}
                 paths:
                     - ~/.cache/yarn
 
+            # cache our node_modules
             - save_cache:
                 key: node-v1-{{ checksum "package.json" }}-{{ arch }}
                 paths:
-                - node_modules
+                    - node_modules
 
-            - persist_to_workspace:
-                root: ~/textile-mobile
+            # create a cache entry which will avoid checking out all 5GB of subsequent builds
+            - save_cache:
+                key: source-v1-{{ .Branch }}-{{ .Revision }}
                 paths:
-                    - node_modules
+                    - ".git"
```

Comment:
here's where our repo is cached for all future builds. (and later stages of this build)

lint / tests go right of the old cache so they can just run in parallel